ipmininet/ipnet.py

```diff
@@ -549,6 +550,64 @@ def ping6Pair(self):
            return: ploss packet loss percentage"""
         return self.pingPair(use_v4=False)
 
+   
+    def runFailurePlan(self,failure_plan):
+        """this function run a failure plan
+            the plan have to be a list of tuple with the name of routeur between
+            which the link have to be down
+            example of plan:
+            [("R1","R2"),("R8","R9"),("R3","R1")]
+            This will return a list of IPIntf 
+        """
+        print("** Starting failure plan")
+        interfaces_down = []
+        for link in failure_plan:
+            try:
+                node1 = self.get(link[0])
+                node2 = self.get(link[1])
+                interfaces = node1.connectionsTo(node2)
+                interfaceNode1 = interfaces[0][0]
+                interfaceNode2 = interfaces[0][1]
+                commande = "ip link set dev " + str(interfaceNode1) + " down"
+                node1.cmd(commande)
+                interfaces_down.append(interfaceNode1)
+                print("** link between " + link[0] + " and "+ link[1] + " down")
+            except Exception as e:
+                log.output("ERROR: node "+ str(e) +"\n")
+                return interfaces_down
+        return interfaces_down
+
+    def restoreLink(self,interfaces):
+        """function which restore the link
+           the argument is a list of IPInft object 
+        """
+        print("** starting restoring link")
+        for interface in interfaces:
+            commande = "ip link set dev " + str(interface) + " up"
+            interface.node.cmd(commande)
+            print("** interfaces " + str(interface) + " up")
+
+    def RandomFailure(self,n):
```

Comment:
You could add an optional parameter that restrict the links that can fail to match https://github.com/cnp3/ipmininet/pull/90#discussion_r492015364

So if the parameter is defined, you use it instead of self.links and otherwise, you stay with self.links :)